### PR TITLE
Switch to desugared logger.

### PR DESCRIPTION
**Pull request by Harwayne**
Switch to desugared logger.

There were actual problems with the PubSub receive adapter. Switched webhook, as well, even though there weren't any issues.

### CODE REVIEW

File: cmd/pubsub/receive_adapter/main.go [hunk 1]
```diff
@@ -53,14 +53,15 @@ func main() {
 		}
 	}
 
-	logger, _ := logging.NewLoggerFromConfig(loggingConfig, component)
+	sl, _ := logging.NewLoggerFromConfig(loggingConfig, component)
+	logger := sl.Desugar()
 	defer flush(logger)
-	ctx := logging.WithLogger(signals.NewContext(), logger)
+	ctx := logging.WithLogger(signals.NewContext(), logger.Sugar())
```
Comment (evankanderson) [inline cmd/pubsub/receive_adapter/main.go]: Why not pass in `sl` here?

File: cmd/webhook/main.go [hunk 2]
```diff
@@ -58,9 +58,9 @@ func SharedMain(resourceHandlers map[schema.GroupVersionKind]webhook.GenericCRD)
 	if err != nil {
 		log.Fatal("Error parsing logging configuration:", err)
 	}
-	logger, atomicLevel := logging.NewLoggerFromConfig(config, component)
-	defer logger.Sync()
-	logger = logger.With(zap.String("cloud.google.com/events", component))
+	sl, atomicLevel := logging.NewLoggerFromConfig(config, component)
+	defer sl.Sync()
```
Comment (evankanderson) [inline cmd/webhook/main.go]: Receive Adapter calls `metrics.FlushExporter()` as well as `sl.Sync`. Do we want to do that, here?

**Comment by n3wscott** [comment]
I think you should make this switch from pkg, not from each repo.

File: cmd/pubsub/receive_adapter/main.go (hunk 1 above)
Comment (Harwayne) [reply cmd/pubsub/receive_adapter/main.go]: Done.

My reasoning was that I wanted to pretend that `sl` didn't exist. But it's not a good reason, so switched.

File: cmd/webhook/main.go (hunk 2 above)
Comment (Harwayne) [reply cmd/webhook/main.go]: Done.

**Comment by Harwayne** [comment]
Ping.

**Comment by Harwayne** [comment]
@evankanderson All comments have been addressed, PTAL.

**Pull request merged**